docs/how-to/installation/upgrades-troubleshoot.txt

```diff
@@ -131,3 +131,39 @@ be able to use Determined as long as you use Docker images based on CUDA
 
    docker run --gpus all --rm nvidia/cuda:9.0-runtime nvidia-smi
    docker run --gpus all --rm nvidia/cuda:10.0-runtime nvidia-smi
+
+*****************************
+ Database Migration Failures
+*****************************
+
+.. code::
+
+   Dirty database version A_LONG_NUMBER. Fix and force version.
+
+If you see the above error message, a database migration was likely
+interrupted while running and the database is now in a dirty state.
+
+Make sure you backup the database and temporarily shutdown the master
+before proceeding further.
+
+To fix this error message, locate the up migration with a suffix of
```

Comment:
In general, doesn't this process depend on exactly what the migration is doing? Like if there are two long statements and the first one finishes, they're probably going to need to dig in and extract part of the migration code to run. Maybe this is the closest thing that's worth putting into a general troubleshooting doc, though. Can't exactly teach everybody to debug arbitrary migrations right here.

Also, as a possible alternative process (for the cases when the migration needs to be rerun in its entirety): I would go into the database and check that it matches the pre-migration state, then set the version to the previous version and `dirty` to false and restart the master, so it'll automatically retry that migration and then just continue from there. Feels a bit nicer to me than running a possibly large migration interactively. Think that would make sense to mention?

Reply:
> In general, doesn't this process depend on exactly what the migration is doing?

Yeah, I mean, ideally our migrations should either be idempotent or in a transaction (so in a transaction since the former isn't always possible with e.g. renames). Then this would always work.

> I would go into the database and check that it matches the pre-migration state, then set the version to the previous version and dirty to false and restart the master, so it'll automatically retry that migration and then just continue from there.

I've thought about this exactly; the issue is if the migration failed for some reason (like time limit from some admin, idk, I've seen it once before) then it's just going to fail over and over again this way. You could say do this + don't mess with the DB at all while it is running.

Reply:
The other way isn't perfect either, but it would be useful to include as an alternative, I think. But anyway I'm happy to get an initial version of this section in and we can iterate on it later.

Reply:
Yeah, if may be useful as like "if the migration was unintentionally interrupted by the user or failed for some other reason known to be intermittent, try that method, else proceed with a more manual approach.